pilot/pkg/model/xds_cache.go

```diff
@@ -94,10 +97,22 @@ func NewXdsCache() XdsCache {
 }
 
 func (x XdsCacheImpl) Run(stop <-chan struct{}) {
-	x.cds.Run(stop)
-	x.eds.Run(stop)
-	x.rds.Run(stop)
-	x.sds.Run(stop)
+	interval := features.XDSCacheIndexClearInterval
+	go func() {
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				go x.cds.flush()
```

Comment:
not sure we should `go` them, in obscure case we could have high lock contention and have many of these stacked up waiting at once...?

If we block, then if the flush() is slow it will exert backpressure on the ticker which is desired

Reply:
lead to unbounded go threads

Reply:
ok. Will change